### PR TITLE
add checking for input values in GANHead constructor

### PR DESCRIPTION
**Pull request by alexpantyukhin**

### CODE REVIEW

**Comment by caisq** [comment]
@joel-shor Can you please review this PR?

File: tensorflow/contrib/gan/python/estimator/python/head_impl.py [hunk 1]
```diff
@@ -90,9 +91,24 @@ def __init__(self, generator_loss_fn, discriminator_loss_fn,
...
+      raise TypeError('generator_loss_fn must be callable.')
+    if not callable(discriminator_loss_fn):
+      raise TypeError('discriminator_loss_fn must be callable.')
+    if not isinstance(generator_optimizer, optimizer.Optimizer):
+      raise TypeError('generator_optimizer must be Optimizer.')
+    if not isinstance(discriminator_optimizer, optimizer.Optimizer):
+      raise TypeError('discriminator_optimizer must be Optimizer.')
+    if not use_loss_summaries in [True, False, None]:
+      raise ValueError('use_loss_summaries must be True, False or None.')
+    if is not None and not callable(get_hooks_fn):
```
Comment (joel-shor) [inline tensorflow/contrib/gan/python/estimator/python/head_impl.py]: `if get_hooks_fn is not None`?

**Comment by alexpantyukhin** [comment]
@joel-shor Fixed the `get_hooks_fn` condition.

**Comment by caisq** [comment]
@joel-shor Please take another look
@tensorflow-jenkins test this please

**Comment by alexpantyukhin** [comment]
@tensorflow-jenkins test this please

**Comment by alexpantyukhin** [comment]
what about this PR?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by alexpantyukhin** [comment]
Is it okay to close the PR?

**Comment by alexpantyukhin** [comment]
Sorry, closed it by mistake.

**Comment by alexpantyukhin** [comment]
@joel-shor  Could you please review the PR?

**Pull request closed without merging**